app/backend/approaches/chatreadretrieveread.py

```diff
@@ -153,29 +155,39 @@ async def run_until_final_call(
             }
         ]
 
+        # Process results
+        current_model = self.hf_model if self.hf_model else self.chatgpt_model
+
+        # Load the Prompty object
+        query_prompty_path = SUPPORTED_MODELS.get(current_model) / "query.prompty"
+        query_prompty = Prompty.load(source=query_prompty_path)
```

Comment:
The receival of the path from SUPPORTED_MODELS is pretty straightforward, I would just add it as a source instead of separately assigning it to the query_prompty_path
```suggestion
        query_prompty_path = SUPPORTED_MODELS.get(current_model) / "query.prompty"
        query_prompty = Prompty.load(source=SUPPORTED_MODELS.get(current_model) / "query.prompty")
```